### PR TITLE
Handle missing `type` key in schema

### LINKED ISSUE #61: The parser crashes if no schema type is specified in the OpenAPI document
**Describe the bug**

The OpenAPI 3 specification prescribes that when no `type` is specified for a schema, this implicitly is equivalent to `anyOf` with all available types (see https://swagger.io/docs/specification/data-models/data-types/#any).

**To Reproduce**
1. Create an OpenAPI document with a schema with the following definition:
    ```yaml
    SomeSchema:
      description: Can be any value
    ```
2. Try to parse this schema
3. Observe a `KeyError` due to the missing `type` key

**Expected behavior**

The parser does not crash and correctly infers the `AnyOf` data type.

**System details (please complete the following information):**
 - OS: Ubuntu
 - OpenAPI 3
 - Python 3.10

**Additional context**

Mostly reporting the findings of https://github.com/maykinmedia/openapi3-parser/pull/1/ to upstream.

### PR DESCRIPTION
**Pull request by sergei-maertens**
Fixes #61

I'm not sure if the approach of setting `schema_type = DataType.ANY_OF` is correct, perhaps this should become an explicit schema with all possible types?

### CODE REVIEW

File: src/openapi_parser/builders/schema.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def create(self, data: dict) -> Schema:
         try:
             schema_type = data['type']
         except KeyError:
-            raise ParserError("Schema does not contain 'type' property")
+            schema_type = DataType.ANY_OF
```
Comment (manchenkoff) [inline src/openapi_parser/builders/schema.py:114]: Would be nice to have some some log entry about implicit type assignment, maybe warning with the previous exception text or similar?

File: tests/builders/schema/test_anyof.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 from openapi_parser.builders.schema import SchemaFactory
 from openapi_parser.enumeration import IntegerFormat
-from openapi_parser.specification import DataType, Discriminator, Integer, AnyOf, String, StringFormat
+from openapi_parser.specification import DataType, Integer, AnyOf, String, StringFormat, Number, Boolean, Array, Object
```
Comment (manchenkoff) [inline tests/builders/schema/test_anyof.py:5]: looks like those imports are not used in this file, can be omitted

File: tests/builders/schema/test_anyof.py [hunk 3]
```diff
@@ -38,6 +38,16 @@
...
     ),
+    (
+        {
+            "description": "Can be any value - string, number, boolean, array or object."
+        },
+        AnyOf(
+            type=DataType.ANY_OF,
+            description="Can be any value - string, number, boolean, array or object.",
+            schemas=[]
+        )
```
Comment (manchenkoff) [inline tests/builders/schema/test_anyof.py:49]: would be nice to have one more case with some actual schema but w/o `type`, for instance some object or array with items

**Review by manchenkoff — CHANGES_REQUESTED** [review]
Left some comments, looking forward to reply 👍

File: tests/builders/schema/test_anyof.py (hunk 3 above)
Comment (SilviaAmAm) [reply tests/builders/schema/test_anyof.py:49]: Do you mean a test case along these lines:
```python
(
        {
            "description": "Array with implicit type.",
            "items": [
                {"type": "string"}
            ]
        },
        AnyOf(
            type=DataType.ANY_OF,
            description="Array with implicit type.",
            schemas=[]
        )
    )
 ```
Or do you mean that in the case with arrays/objects the type should be inferred? Reading this SO answer: https://stackoverflow.com/questions/47374980/schema-object-without-a-type-attribute-in-swagger-2-0/47390723#47390723 it seems that we maybe should not infer types.  What do you think?

**Review by manchenkoff — APPROVED** [review]

**Comment by manchenkoff** [comment]
Thanks, published as v1.1.11 version 👍

**Comment by sergei-maertens** [comment]
Thank you! 🙌

**Pull request merged**